hmf-common/src/main/java/com/hartwig/hmftools/common/utils/file/DelimFileReader.java

```diff
@@ -293,272 +294,143 @@ private Row(final Map<String, Integer> columnIndexMap, final String[] values)
+        @Nullable public Integer getIntOrNull(Enum<?> column) { return getInt(getColumn(column), true); }
 
-        public boolean isNull(final Enum<?> column)
-        {
-            return isNull(column.name());
-        }
+        public long getLong(String column) { return getLong(getColumn(column), false); }
+        public long getLong(Enum<?> column) { return getLong(getColumn(column), false); }
+        @Nullable public Long getLongOrNull(String column) { return getLong(getColumn(column), true); }
+        @Nullable public Long getLongOrNull(Enum<?> column) { return getLong(getColumn(column), true); }
 
-        public String get(final Enum<?> column)
-        {
-            return get(column.name());
-        }
+        public float getFloat(String column) { return getFloat(getColumn(column), false); }
+        public float getFloat(Enum<?> column) { return getFloat(getColumn(column), false); }
+        @Nullable public Float getFloatOrNull(String column) { return getFloat(getColumn(column), true); }
+        @Nullable public Float getFloatOrNull(Enum<?> column) { return getFloat(getColumn(column), true); }
 
-        public @Nullable String getOrNull(final Enum<?> column)
-        {
-            return getOrNull(column.name());
-        }
+        public double getDouble(int column) { return getDouble(column, false); }
+        public double getDouble(String column) { return getDouble(getColumn(column), false); }
+        public double getDouble(Enum<?> column) { return getDouble(getColumn(column), false); }
+        @Nullable public Double getDoubleOrNull(String column) { return getDouble(getColumn(column), true); }
+        @Nullable public Double getDoubleOrNull(Enum<?> column) { return getDouble(getColumn(column), true); }
 
-        public int getInt(final Enum<?> column)
+        private String getString(int column, boolean allowNull)
         {
-            return getInt(column.name());
+            return getValue(column, Function.identity(), allowNull);
         }
 
-        public @Nullable Integer getIntOrNull(final Enum<?> column)
+        private Byte getByte(int column, boolean allowNull)
         {
-            return getIntOrNull(column.name());
+            return getValue(column, Byte::parseByte, allowNull);
         }
 
-        public boolean getBoolean(final Enum<?> column)
+        private Character getChar(int column, boolean allowNull)
         {
-            return getBoolean(column.name());
+            // TODO? should forbid more than 1 char
+            return getValue(column, string -> string.charAt(0), allowNull);
         }
 
-        public @Nullable Boolean getBooleanOrNull(final Enum<?> column)
+        private Boolean getBoolean(int column, boolean allowNull)
         {
-            return getBooleanOrNull(column.name());
+            // TODO: should accept only true or false
+            return getValue(column, Boolean::parseBoolean, allowNull);
```

Comment:
yes we should make sure this gets fixed. Possibly by making our own `parseBoolean` function here.